woocommerce/woocommerce-ios
# [IPPInAppFeedback] UI: Payments banner placement

Part of https://github.com/woocommerce/woocommerce-ios/issues/8530

## Description

This PR adds the in-app feedback banner in the Payments view, as well as the buttons to launch a survey and dismiss the banner. 

At the moment the survey is only dismissible when tapping on "dismiss" (not upon survey completion), and will show up again each time we navigate to the Payments view.

## TODO:
- [ ] Design has changed. Needs to be re-implemented.
- [ ] Dismissal and show me later/don't show again UI will be added on a different PR. Most probably using `FeatureAnnouncementCardView`

## Changes
- We reuse the existing `Notice` component to create a feedback notice banner in the Payments View.
- Extended Notice to accept a new `noticeTappedHandler` handler noticeTappedHandler, so we can use 2 different behaviours: When the notice is tapped (show survey), and when the action button is tapped (dismiss notice).
- Added a new `IPPFeedback` case to `SurveyViewController` which will launch and manage the Survey launched via the tapping the notice
- Added a temporary testing survey link ( `https://automattic.survey.fm/woo-app-ipp-in-app-feedback-testing` ), this will dynamically change in future PRs based on certain conditions.


## Testing instructions
1. Go to Menu > Payments > See the notice on the bottom of the screen
2. Tap the notice, a WebView showing a survey form will shop up. Scroll down and tap "Finish Survey", a "Feedback Sent" view will be shown and you'll be redirected to the Payments page.
3. Navigate to a different View, then navigate to Menu > Payments again. 
4. Tap on "X" and see the banner disappear.

## Screenshots

| Feedback Banner | Give Feedback  | Feedback Sent  | 
|---|---|---|
| <img width=400 src="https://user-images.githubusercontent.com/3812076/211466952-7152a1b2-0caa-4e5e-8ed1-099e98d2d8e7.png"> | <img width=400 src="https://user-images.githubusercontent.com/3812076/210490819-afeb11b2-95f2-4e93-b018-0f4468d971d1.png"> | <img width=400 src="https://user-images.githubusercontent.com/3812076/210490830-4cf628a7-78ef-4e70-b8d5-73cd553a9e69.png"> |
